File: Isomorphic Strings.py
# Link: https://leetcode.com/problems/isomorphic-strings/

class Solution(object):
    def isIsomorphic(self, s, t):
        st = {}
        ts = {}

        for c1, c2 in zip(s, t):
            if (c1 in st and st[c1] != c2) or (c2 in ts and ts[c2] != c1):
                return False
                
            st[c1] = c2
            ts[c2] = c1

        return True

File: isomorphic-strings.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        unordered_map<char, char>mpS, mpT;
        int l = s.size();
        for(int i = 0; i < l; i++){
            mpS[s[i]] = t[i];
            mpT[t[i]] = s[i];
        }

        for(int i = 0; i < l; i++){
            if(mpS[s[i]] != t[i] || mpT[t[i]] != s[i])
            return false;
        }

        return true;
    }
};